tools/render-test/render-d3d11.cpp

```diff
@@ -201,102 +195,88 @@ Renderer* createD3D11Renderer()
 
     // The `D3DCompile` entry point takes a bunch of parameters, but we
     // don't really need most of them for Slang-generated code.
-    ID3DBlob* dxShaderBlob = nullptr;
-    ID3DBlob* dxErrorBlob = nullptr;
-    HRESULT hr = D3DCompile_(
-        source,
-        strlen(source),
-        sourcePath,
-        &defines[0],
-        nullptr,
-        entryPointName,
-        dxProfileName,
-        flags,
-        0,
-        &dxShaderBlob,
-        &dxErrorBlob);
+    ID3DBlob* shaderBlob = nullptr;
+    ID3DBlob* errorBlob = nullptr;
+    HRESULT hr = compileFunc(source, strlen(source), sourcePath, &defines[0], nullptr, entryPointName, dxProfileName, flags, 0,
+        &shaderBlob, &errorBlob);
 
     // If the HLSL-to-bytecode compilation produced any diagnostic messages
     // then we will print them out (whether or not the compilation failed).
-    if (dxErrorBlob)
+    if (errorBlob)
     {
-        fputs(
-            (char const*)dxErrorBlob->GetBufferPointer(),
-            stderr);
-        fflush(stderr);
-
-        OutputDebugStringA(
-            (char const*)dxErrorBlob->GetBufferPointer());
+        ::fputs((const char*)errorBlob->GetBufferPointer(), stderr);
```

Comment:
Do you advocate for prefixing all calls to global-scope functions with `::`? I usually would only do that if there was a serious risk of collision with symbols defined in the enclosing class/namespace.

Reply:
I think it makes it clear where the function is coming from. Within other code bases such calls are by design very unusual - so this extra visual mark, highlights the use.

Reply:
There’s a good point there that Slang currently calls directly into stuff like `fopen` right now, which will make it harder to integrate into a lot of engines/applications (which need/expect to mediate all operations like that).

For testing code like this it is less important, but if we go to the trouble of implement a file system abstraction later for Slang proper, we might just use it here anyway.